tests/model/test_model.py

```diff
@@ -3641,10 +3641,10 @@ def test_is_muted_topic(
                 (1, "topic"),
                 id="unread_present_before_previous_topic",
             ),
-            case(  # TODO Should be None? (2 other cases)
+            case(
                 {(1, "topic")},
                 (1, "topic"),
-                (1, "topic"),
+                None,
```

Comment:
I guess this does not represent the case you're talking about, since the current function only determines the next topic to narrow to, and not the next message. The expected behavior of the 'n' hotkey only seems to be to narrow to the next topic with unread messages, and it is not necessary read through all unread messages in the current topic before doing so.
 
Explore mode is an interesting feature. As I mentioned above about the expected behavior of the 'n' hotkey, not reading the messages should not affect it, I suppose. Please correct me if I'm wrong in my understanding of the expected behavior.

Reply:
Right, this is fine, I was simply surprised reading it :)